tools/src/main/python/dlpx/virtualization/_internal/codegen.py

```diff
@@ -90,10 +90,34 @@ def generate_python(name, source_dir, plugin_config_dir, schema_content):
     _copy_generated_to_dir(output_dir, source_dir)
 
 
+#
+# Makes all references to platform json schema definitions "opaque" by replacing
+# them with plain object types.
+#
+def _make_url_refs_opaque(json):
+    if isinstance(json, dict):
+        for key in json:
+            if key == '$ref' and isinstance(json[key], basestring)\
```

Comment:
can you just unit this test this? i think you could probably put something in test_codegen.py. let me know if you need me to walk you through the testing framework.